services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py

```diff
@@ -0,0 +1,130 @@
+""" Handles request to the viewers redirection entrypoints
+
+"""
+import logging
+from typing import Optional
+
+import aiohttp
+from aiohttp import web
+from aiohttp.client_exceptions import ClientError
+from pydantic import BaseModel, HttpUrl, ValidationError
+from pydantic.types import PositiveInt
+from yarl import URL
+
+from ..statics import INDEX_RESOURCE_NAME
+from ._core import (
+    MatchNotFoundError,
+    ValidationMixin,
+    ViewerInfo,
+    find_compatible_viewer,
+)
+from ._projects import acquire_project_with_viewer
+from ._users import UserInfo, acquire_user, ensure_authentication
+
+log = logging.getLogger(__name__)
+
+
+def create_redirect_response(
+    app: web.Application, page: str, **parameters
+) -> web.HTTPFound:
+    """
+    Returns a redirect response to the front-end with information on page and parameters embedded in the fragment.
+
+    For instance,
+        https://osparc.io/#/error?message=Sorry%2C%20I%20could%20not%20find%20this%20&status_code=404
+    results from
+            - page=error
+        and parameters
+            - message="Sorry, I could not find this"
+            - status_code=404
+    """
+    page = page.strip(" /")
+    # TODO: test that fragment queries are understood by front-end
+    # TODO: front end should create an error page and a view page
+    assert page in ("view", "error")  # nosec
+
+    in_fragment = str(URL.build(path=f"/{page}").with_query(**parameters))
+    redirect_url = app.router[INDEX_RESOURCE_NAME].url_for().with_fragment(in_fragment)
+    return web.HTTPFound(location=redirect_url)
+
+
+# HANDLERS --------------------------------
+
+
+class QueryParams(BaseModel, ValidationMixin):
+    # TODO: create dinamically with pydantic class
+    file_name: Optional[str] = None
+    file_size: PositiveInt
+    file_type: str  # TODO: should we define some types?
+    download_link: HttpUrl
+
+    async def check_download_link(self):
+        """Explicit validation of download link that performs a light fetch of url's hea"""
+        #
+        # WARNING: Do not use this check with Amazon download links
+        #          since HEAD operation is forbidden!
+        try:
+            async with aiohttp.request("HEAD", self.download_link) as response:
```

Comment:
Please put a small timeout here. Will avoid hanging requests due to site having issues responding etc...
You might also want to customise the HTTP error in case of timeout.